src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BindLoweringPass.cs

```diff
@@ -0,0 +1,419 @@
+            {
+                return false;
+            }
+
+            // The the tag helper specifies attribute names, they should win.
+            //
+            // This handles cases like <input type="text" bind="@Foo" /> where the tag helper is 
+            // generated to match a specific tag and has metadata that identify the attributes.
+            //
+            // We expect 1 bind tag helper per-node.
+            var bindTagHelper = node.TagHelpers.Single(t => t.IsBindTagHelper());
+            valueAttributeName = bindTagHelper.GetValueAttributeName() ?? valueAttributeName;
+            changeHandlerAttributeName = bindTagHelper.GetChangeHandlerAttributeName() ?? changeHandlerAttributeName;
+
+            // We expect 0-1 components per-node.
+            var componentTagHelper = node.TagHelpers.FirstOrDefault(t => t.IsComponentTagHelper());
+            if (componentTagHelper == null)
+            {
+                // If it's not a component node then there isn't too much else to figure out.
+                return attributeName != null && changeHandlerAttributeName != null;
+            }
+
+            // If this is a component, we need an attribute name for the value.
+            if (attributeName == null)
+            {
+                return false;
+            }
+
+            // If this is a component, then we can infer '<PropertyName>Changed' as the name
+            // of the change handler.
+            if (changeHandlerAttributeName == null)
+            {
+                changeHandlerAttributeName = valueAttributeName + "Changed";
+            }
+
+            for (var i = 0; i < componentTagHelper.BoundAttributes.Count; i++)
+            {
+                var attribute = componentTagHelper.BoundAttributes[i];
+
+                if (string.Equals(valueAttributeName, attribute.Name))
+                {
+                    valueAttribute = attribute;
+                }
+
+                if (string.Equals(changeHandlerAttributeName, attribute.Name))
+                {
+                    changeHandlerAttribute = attribute;
+                }
+            }
+
+            return true;
+        }
+
+        private bool TryGetFormatNode(
+            TagHelperIntermediateNode node,
+            ComponentAttributeExtensionNode attributeNode,
+            string valueAttributeName,
+            out ComponentAttributeExtensionNode formatNode)
+        {
+            for (var i = 0; i < node.Children.Count; i++)
+            {
+                var child = node.Children[i] as ComponentAttributeExtensionNode;
+                if (child != null &&
+                    child.TagHelper != null &&
+                    child.TagHelper == attributeNode.TagHelper &&
+                    ((child.AttributeName == "format" && attributeNode.AttributeName == "bind") ||
+                    (child.AttributeName == "format-" + valueAttributeName)))
```

Comment:
it it possible we could pick something that doesn't share a prefix with `bind-`? This degrades the UX of the fallback cases - `bind-value-onchange` and `bind-format-value` are the *same thing* from tooling's point of view. We do whatever we want for code generation, but intellisense and completion are fairly primitive.

We could also consider doing something like `value-format` and just decoupling `format` from `bind` totally. I'm pretty sure we have the ability to do matching based on a suffix.

Reply:
Oh OK that's a good point. From the point of view of not clashing with native web standards, anything with a hyphen is very likely sufficient based on historical precedent. Whatever you think makes most sense and works best in tooling, and also contains a hyphen, would be fine with me!

`format-value` or `value-format` (particularly the latter) sound good to me.

Reply:
OK, I'm going to try `value-format`, and keeping it coupled to `bind-...` for now. 

So what that means is that you'd have to write `<input type="date" bind="@SelectedDate" value-format="MM/dd/yyyy" />` - you have to know what the attribute is that you're formatting. Do you think that makes sense?

Reply:
@jhu78 

> I think that could be confusing. How would I format the content of a textarea? 

If I'm not mistaken, that would look like `<textarea bind="@MyValue" value-format="...">`

I think you might be expecting a level of sophistication from this **initial** feature add that won't be here for some time. 

> Actually, I didn't mind the @Bind notation. But since you removed the leading @ why keep it at all? Wouldn't bind="CurrentValue" be enough?

This feature is replacing something that was hacked in for prototype with something much more Razor-like, with a much better tooling experience.  

Yes, you're correct that `bind="CurrentValue"` would work as well. I suppose that it's an old habit of mind to be explicit 😆